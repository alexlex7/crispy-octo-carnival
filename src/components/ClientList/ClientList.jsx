import React from 'react';
import ClientItem from '../ClientItem/ClientItem';
import styles from './ClientList.module.css';
import PropTypes from 'prop-types';

export default function ClientList({ items, isMobile = false }) {
  const clients = isMobile ? items.slice(0, 5) : items.slice(0, 6);
  // console.log(cli
  return (
    <ul className={isMobile ? styles.mobileList : styles.desktopList}>
      {clients.map((client) => (
        <ClientItem key={client.id} client={client} isMobile={isMobile} />
      ))}
    </ul>
  );
}

ClientList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      imageURL: PropTypes.string.isRequired,
      mobileImageURL: PropTypes.string.isRequired,
    })
  ),
  isMobile: PropTypes.bool,
};
