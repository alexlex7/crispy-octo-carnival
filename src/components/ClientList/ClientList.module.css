.mobileList {
  display: flex;
  column-gap: 28px;
  list-style: none;
  padding: 0;
  margin: 0;
}

.desktopList {
  display: flex;
  column-gap: 100px;
  list-style: none;
  padding: 0;
  margin: 0;
}
