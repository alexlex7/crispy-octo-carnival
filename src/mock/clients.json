[
  {
    "id": "1",
    "name": "Mastercard",
    "imageURL": "http://localhost:8080/images/clients/masterCard.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/masterCardMobile.jpg"
  },
  {
    "id": "2",
    "name": "Airbnb",
    "imageURL": "http://localhost:8080/images/clients/airbnb.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/airbnbMobile.jpg"
  },
  {
    "id": "3",
    "name": "Uber",
    "imageURL": "http://localhost:8080/images/clients/uber.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/uberMobile.jpg"
  },
  {
    "id": "4",
    "name": "PayPal",
    "imageURL": "http://localhost:8080/images/clients/paypal.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/paypalMobile.jpg"
  },
  {
    "id": "5",
    "name": "Visa",
    "imageURL": "http://localhost:8080/images/clients/visa.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/visaMobile.jpg"
  },
  {
    "id": "6",
    "name": "Stripe",
    "imageURL": "http://localhost:8080/images/clients/stripe.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/visaMobile.jpg"
  }
]
