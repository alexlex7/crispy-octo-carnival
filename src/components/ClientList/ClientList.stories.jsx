import React from 'react';
import ClientList from './ClientList';
import clients from '../../mock/clients.json';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Clients/ClientList',
  component: ClientList,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <ClientList {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  items: clients,
};
