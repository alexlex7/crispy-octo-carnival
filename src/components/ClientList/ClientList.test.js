import { render, screen } from '@testing-library/react';
import ClientList from './ClientList';
import clients from '../../mock/clients.json';

test('rendes clients list component', () => {
  const { asFragment } = render(<ClientList items={clients} />);
  const linkElement = screen.getByRole('list');
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
