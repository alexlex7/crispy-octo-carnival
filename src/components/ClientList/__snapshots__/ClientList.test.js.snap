// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendes clients list component 1`] = `
<DocumentFragment>
  <ul
    class="desktopList"
  >
    <li
      class="listItem"
    >
      <img
        alt="Mastercard"
        src="http://localhost:8080/images/clients/masterCard.jpg"
        title="Mastercard"
      />
    </li>
    <li
      class="listItem"
    >
      <img
        alt="Airbnb"
        src="http://localhost:8080/images/clients/airbnb.jpg"
        title="Airbnb"
      />
    </li>
    <li
      class="listItem"
    >
      <img
        alt="Uber"
        src="http://localhost:8080/images/clients/uber.jpg"
        title="Uber"
      />
    </li>
    <li
      class="listItem"
    >
      <img
        alt="PayPal"
        src="http://localhost:8080/images/clients/paypal.jpg"
        title="PayPal"
      />
    </li>
    <li
      class="listItem"
    >
      <img
        alt="Visa"
        src="http://localhost:8080/images/clients/visa.jpg"
        title="Visa"
      />
    </li>
    <li
      class="listItem"
    >
      <img
        alt="Stripe"
        src="http://localhost:8080/images/clients/stripe.jpg"
        title="Stripe"
      />
    </li>
  </ul>
</DocumentFragment>
`;
